src/app/problems/_services/question.service.spec.ts

```diff
@@ -2,19 +2,126 @@ import {TestBed} from '@angular/core/testing';
 
 import {QuestionService} from './question.service';
 import {TestModule} from '@test/test.module';
+import {ApiService} from "@app/_services/api.service";
+import {HttpTestingController} from "@angular/common/http/testing";
+import {Question} from "@app/_models";
+import {
+    MOCK_CHECKBOX_QUESTION,
+    MOCK_JAVA_FORM_DATA,
+    MOCK_MCQ_FORM_DATA,
+    MOCK_MCQ_QUESTION,
+    MOCK_PARSONS_FORM_DATA
+} from "@test/mock";
 
 describe('QuestionService', () => {
-    let service: QuestionService;
+    let mockQuestions: Question[];
+    let questionService: QuestionService;
+    let apiService: ApiService;
+    let httpMock: HttpTestingController;
 
     beforeEach(() => {
+        mockQuestions = [MOCK_MCQ_QUESTION, MOCK_CHECKBOX_QUESTION];
         TestBed.configureTestingModule({
             imports: [TestModule],
-            providers: [QuestionService]
+            providers: [QuestionService, ApiService]
         });
-        service = TestBed.inject(QuestionService);
+        questionService = TestBed.inject(QuestionService);
+        apiService = TestBed.inject(ApiService);
+        httpMock = TestBed.inject(HttpTestingController);
+    });
+
+    afterEach(() => {
+        httpMock.verify();
     });
 
     it('should be created', () => {
-        expect(service).toBeTruthy();
+        expect(questionService).toBeTruthy();
+    });
+
+    it('getQuestions returns all questions', () => {
+        questionService.getQuestions().subscribe((questions) => {
+            expect(questions.results.length).toEqual(2);
+            expect(questions.results).toEqual(mockQuestions);
+        });
+        const request = httpMock.expectOne('http://localhost:8000/api/questions/?page=1&page_size=50&search=&category__parent__name=&category__name=&difficulty=&is_sample=&ordering=');
+        expect(request.request.method).toBe('GET');
+        request.flush(mockQuestions);
+    });
+
+    it('getQuestion returns a single question', () => {
+        questionService.getQuestion(0).subscribe((question) => {
+            expect(question.id).toEqual(0);
+            expect(question.is_checkbox).toBeFalsy();
+        });
+        const request = httpMock.expectOne(apiService.getURL('questions', 0));
+        expect(request.request.method).toBe('GET');
+        request.flush(mockQuestions[0]);
```

Comment:
it's better to gave generic tests meaning that the test should not break if we change the mock questions.
start with `const question = mockQuestions[0]` and then use `question` and `question.id` instead of `0` and `mockQuestions[0]`.